### PR TITLE
sensei: add versions and fix dependencies

### PR DESCRIPTION
**Pull request by danlipsa**
@devel should work with all options.

Releases don't work with ~catalyst
See https://gitlab.kitware.com/sensei/sensei/-/merge_requests/240 for the fix for develop.

Releases work only with paraview 5.7 and 5.6
See https://gitlab.kitware.com/sensei/sensei/-/merge_requests/239 for the fix for develop (which works with 5.9)

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed new versions available at the URL.

Approval pending feedback from package maintainer and all required checks passing.

File: var/spack/repos/builtin/packages/sensei/package.py [hunk 1]
```diff
@@ -87,7 +93,8 @@ def cmake_args(self):
         if '+libsim' in spec:
             args.extend([
                 '-DENABLE_LIBSIM:BOOL=ON',
-                '-DVISIT_DIR:PATH={0}'.format(spec['visit'].prefix),
+                '-DVISIT_DIR:PATH={}/current/{}-{}'.format(
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sensei/package.py]: FWIW. This is likely the line causing the validation check problem.  I recommend adding numbers within the curly braces (starting with '0').

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sensei/package.py]: Thanks for the suggestion. I'll do that.

**Comment by danlipsa** [comment]
@chuckatkins Please review.

**Comment by chuckatkins** [comment]
Just marked as "dont-merge-yet" while we test the multitude of configurations this needs to work with (not saying there's anything wrong with it, just that it's still being vetted)

File: var/spack/repos/builtin/packages/sensei/package.py [hunk 2]
```diff
@@ -41,20 +43,23 @@ class Sensei(CMakePackage):
...
+    depends_on("visit~gui~python", when="+libsim")
+    depends_on("vtk@8.1.0:8.1.2", when="+libsim")
     depends_on("vtk", when="~libsim ~catalyst")
+    depends_on("vtk+python", when="~libsim ~catalyst+python")
     depends_on("adios", when="+adios")
-    depends_on("hdf5", when="+hdf5")
+    # VTK needs +hl and currently spack cannot resolve +hl and ~hl
+    depends_on("hdf5+hl", when="+hdf5")
     # SENSEI 3 supports Python 3, earlier versions upport only Python 2
     depends_on("python@:2.7.16", when="@:2.1.1 +python", type=('build', 'run'))
```
Comment (chuckatkins) [inline var/spack/repos/builtin/packages/sensei/package.py:64]: You'll also need to add the following to make sure the python runtime is appropriately setup:
`    extends('python', when='+python')`

Comment (danlipsa) [reply var/spack/repos/builtin/packages/sensei/package.py:64]: Thanks Chuck. I added that.

**Review by chuckatkins — DISMISSED** [review]

**Review by chuckatkins — DISMISSED** [review]

**Review by chuckatkins — APPROVED** [review]

**Pull request merged**